instrumentation/kamon-pekko-http/src/main/scala-2.12/kamon/instrumentation/pekko/http/PekkoHttpServerInstrumentation.scala

```diff
@@ -0,0 +1,349 @@
+
+import scala.collection.immutable
+
+
+class PekkoHttpServerInstrumentation extends InstrumentationBuilder {
+
+  /**
+    * When instrumenting bindAndHandle what we do is wrap the Flow[HttpRequest, HttpResponse, NotUsed] provided by
+    * the user and add all the processing there. This is the part of the instrumentation that performs Context
+    * propagation, tracing and gather metrics using the HttpServerInstrumentation packed in common.
+    *
+    * One important point about the HTTP Server instrumentation is that because it is almost impossible to have a proper
+    * operation name before the request processing hits the routing tree, we are delaying the sampling decision to the
+    * point at which we have some operation name.
+    */
+
+  onType("org.apache.pekko.http.scaladsl.HttpExt")
+    .advise(method("bindAndHandle"), classOf[HttpExtBindAndHandleAdvice])
+
+  /**
+    * For the HTTP/2 instrumentation, since the parts where we can capture the interface/port and the actual flow
+    * creation happen at different times we are wrapping the handler with the interface/port data and reading that
+    * information when turning the handler function into a flow and wrapping it the same way we would for HTTP/1.
+    */
+
+  onType("org.apache.pekko.http.impl.engine.http2.Http2Ext")
+    .advise(method("bindAndHandleAsync") and isPublic(), classOf[Http2ExtBindAndHandleAdvice])
+
+  onType("org.apache.pekko.http.impl.engine.http2.Http2Blueprint$")
+    .intercept(method("handleWithStreamIdHeader"), Http2BlueprintInterceptor)
+
+  /**
+    * The rest of these sections are just about making sure that we can generate an appropriate operation name (i.e. free
+    * of variables) and take a Sampling Decision in case none has been taken so far.
+    */
+  onType("org.apache.pekko.http.scaladsl.server.RequestContextImpl")
+    .mixin(classOf[HasMatchingContext.Mixin])
+    .intercept(method("copy"), RequestContextCopyInterceptor)
+
+  onType("org.apache.pekko.http.scaladsl.server.directives.PathDirectives")
+    .intercept(method("rawPathPrefix"), classOf[PathDirectivesRawPathPrefixInterceptor])
+
+  onType("org.apache.pekko.http.scaladsl.server.directives.FutureDirectives")
+    .intercept(method("onComplete"), classOf[ResolveOperationNameOnRouteInterceptor])
+
+  onTypes("org.apache.pekko.http.scaladsl.server.directives.OnSuccessMagnet$", "org.apache.pekko.http.scaladsl.server.directives.CompleteOrRecoverWithMagnet$")
+    .intercept(method("apply"), classOf[ResolveOperationNameOnRouteInterceptor])
+
+  onType("org.apache.pekko.http.scaladsl.server.directives.RouteDirectives")
+    .intercept(method("complete"), classOf[ResolveOperationNameOnRouteInterceptor])
+    .intercept(method("redirect"), classOf[ResolveOperationNameOnRouteInterceptor])
+    .intercept(method("failWith"), classOf[ResolveOperationNameOnRouteInterceptor])
+
+  /**
+    * Akka-http 10.1.x compatibility.
+    */
+
+  onType("org.apache.pekko.http.scaladsl.Http2Ext")
```

Comment:
again, pekko-http may not need this - there is just a v1.0.0 to handle (based on akka-http 10.2)

Reply:
Correct! that package is not there in Pekko HTTP. `org.apache.pekko.http.impl.engine.http2.Http2Ext` is, which is instrumented.